### DISCUSSION TITLE
Welcome to lego-hub-tk Discussions!

### DISCUSSION THREAD

**Discussion by smr99** [smr99/lego-hub-tk#24] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by arnecuperus** [comment]
Is it possible to for example control motors with this project?

**Reply by arnecuperus** [reply]
Using the micropython shell maybe?

**Reply by smr99** [reply]
The project is mostly focused on communicating with the hub from a computer.  One aspect of this is to be able to download & run python programs to the hub.  And communicate while the programs run.  So the you can certainly write and download a python program that controls the motors.  But the software in this project is limited to helping you download your code.

**Reply by arnecuperus** [reply]
How did you discover that you can set individual pixels by sending this:  
```python
def display_set_pixel(self, x, y, brightness = 9):
    return self.send_message('scratch.display_set_pixel', { 'x':x, 'y': y, 'brightness': brightness})
```
Is there some sort of online documentation which covers the "scratch" commands?

**Reply by smr99** [reply]
Ah.  The bulk of "run_command.py" -- which includes the cited code -- is copied from https://github.com/nutki/spike-tools.  I don't know how this was discovered.  The spike-tools also had a bunch of code related to run-time communications with the hub that I also incorporated.  My contributions are the rest of the code -- things like communicating via bluetooth and serial with failover.

Last year the LEGO group published the hub python API here: https://lego.github.io/MINDSTORMS-Robot-Inventor-hub-API/
This is a great resource for writing python code to run on the hub.

**Reply by arnecuperus** [reply]
Spike tools was originally mend to be only be used with a USB connection right?

**Reply by arnecuperus** [reply]
After digging around in the firmware (specifically the [commands folder](https://github.com/gpdaniels/spike-prime/tree/master/filesystem/storm%20-%20v1.4.01.0000-594ce3d%20-%203.1.43/commands)) I found the following: 
![Schermafbeelding 2022-04-25 om 11 39 04](https://user-images.githubusercontent.com/90452358/165063627-9d110d47-b6d9-420d-b771-b04a269b21b1.png)
On the right side you can see the methods used in the program from nutki. There are many more files which will probably contain more commands!

**Reply by arnecuperus** [reply]
Take a look at my fork, it has added motor functionality!
